ndcube/tests/test_ndcube.py

```diff
@@ -1,819 +1,277 @@
-                          [1., 1., 1.]],
-                         unit=u.deg),
-              u.Quantity([[1.26915033e-05, 4.99987815e-01, 9.99962939e-01],
-                          [1.26918126e-05, 5.00000000e-01, 9.99987308e-01]], unit=u.deg)),
-             u.Quantity([1.02e-09, 1.04e-09, 1.06e-09, 1.08e-09], unit=u.m))),
-    ((cubem[:, :, 0]), (u.Quantity([1., 1.26918126e-05], unit=u.deg),
-                        u.Quantity([1., 5.00000000e-01], unit=u.deg),
-                        u.Quantity([1., 9.99987308e-01], unit=u.deg)))
-])
-def test_axis_world_coords_values_without_input(test_input, expected):
-    all_coords = test_input.axis_world_coords_values()
-    for i in range(len(all_coords)):
-        assert_quantity_allclose(all_coords[i], expected[i])
-        assert all_coords[i].unit == expected[i].unit
-
-
-@pytest.mark.parametrize("test_input,expected", [
-    ((cubem, 0, 0), ((2 * u.pix, 3 * u.pix, 4 * u.pix), NDCubeSequence, dict, NDCube, OrderedDict)),
-    ((cubem, 1, 0), ((3 * u.pix, 2 * u.pix, 4 * u.pix), NDCubeSequence, dict, NDCube, OrderedDict)),
-    ((cubem, -2, 0), ((3 * u.pix, 2 * u.pix, 4 * u.pix), NDCubeSequence, dict, NDCube, OrderedDict))
-])
-def test_explode_along_axis(test_input, expected):
-    inp_cube, inp_axis, inp_slice = test_input
-    exp_dimensions, exp_type_seq, exp_meta_seq, exp_type_cube, exp_meta_cube = expected
-    output = inp_cube.explode_along_axis(inp_axis)
-    assert tuple(output.dimensions) == tuple(exp_dimensions)
-    assert any(output[inp_slice].dimensions ==
-               u.Quantity((exp_dimensions[1], exp_dimensions[2]), unit='pix'))
-    assert isinstance(output, exp_type_seq)
-    assert isinstance(output[inp_slice], exp_type_cube)
-    assert isinstance(output.meta, exp_meta_seq)
-    assert isinstance(output[inp_slice].meta, exp_meta_cube)
-
-
-def test_array_axis_physical_types():
+
+
+def generate_data(shape):
+    data = np.arange(np.product(shape))
+    return data.reshape(shape)
+
+
+def test_wcs_object(all_ndcubes):
+    assert isinstance(all_ndcubes.wcs.low_level_wcs, BaseLowLevelWCS)
+    assert isinstance(all_ndcubes.wcs, BaseHighLevelWCS)
+
+
+@pytest.mark.parametrize("ndc, item",
+                         (
+                             ("ndcube_3d_ln_lt_l", np.s_[:, :, 0]),
+                             ("ndcube_3d_ln_lt_l", np.s_[..., 0]),
+                             ("ndcube_3d_ln_lt_l", np.s_[1:2, 1:2, 0]),
+                             ("ndcube_3d_ln_lt_l", np.s_[..., 0]),
+                             ("ndcube_3d_ln_lt_l", np.s_[:, :, 0]),
+                             ("ndcube_3d_ln_lt_l", np.s_[1:2, 1:2, 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[:, :, 0, 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[..., 0, 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[1:2, 1:2, 1, 1]),
+                         ),
+                         indirect=("ndc",))
```

Comment:
How does this `indirect` thing work?

Reply:
It is a froody fixture lookup thing. Basically it lets you lookup a fixture by name. It passes the name specified in the parametrize to the `ndc` fixture which returns the fixture with the given name.